composer/datasets/__init__.py

```diff
@@ -1,32 +1,42 @@
-from composer.datasets.mnist import MNISTDatasetHparams as MNISTDatasetHparams
-from composer.datasets.mnist import MNISTWebDatasetHparams as MNISTWebDatasetHparams
-from composer.datasets.synthetic import MemoryFormat as MemoryFormat
-from composer.datasets.synthetic import SyntheticBatchPairDataset as SyntheticBatchPairDataset
-from composer.datasets.synthetic import SyntheticDataLabelType as SyntheticDataLabelType
-from composer.datasets.synthetic import SyntheticDataType as SyntheticDataType
-from composer.datasets.synthetic import SyntheticPILDataset as SyntheticPILDataset
+* All datasets derive from the abstract base class :class:`~.DatasetHparams` and it contains common parameters such
+
+  or a :class:`~.DataSpec`) used by the trainer.
+
+* :class:`~.DataloaderHparams` contains the :class:`torch.utils.data.Dataloader` settings that are common across
+  both training and eval datasets. See the documentation of :class:`~.DataloaderHparams` for more details on these
+  settings.
+"""
+
+from composer.datasets.ade20k import ADE20kDatasetHparams, ADE20kWebDatasetHparams
+from composer.datasets.brats import BratsDatasetHparams
+from composer.datasets.c4 import C4DatasetHparams
+from composer.datasets.cifar import (CIFAR10DatasetHparams, CIFAR10WebDatasetHparams, CIFAR20WebDatasetHparams,
+                                     CIFAR100WebDatasetHparams)
+from composer.datasets.coco import COCODatasetHparams
+from composer.datasets.dataloader import DataloaderHparams, WrappedDataLoader
+from composer.datasets.dataset_registry import get_dataset_registry
+from composer.datasets.evaluator import EvaluatorHparams
+from composer.datasets.glue import GLUEHparams
+from composer.datasets.hparams import DatasetHparams, SyntheticHparamsMixin, WebDatasetHparams
+from composer.datasets.imagenet import (Imagenet1kWebDatasetHparams, ImagenetDatasetHparams,
+                                        TinyImagenet200WebDatasetHparams)
+from composer.datasets.lm_datasets import LMDatasetHparams
+from composer.datasets.mnist import MNISTDatasetHparams, MNISTWebDatasetHparams
+from composer.datasets.synthetic import (MemoryFormat, SyntheticBatchPairDataset, SyntheticDataLabelType,
+                                         SyntheticDataType, SyntheticPILDataset)
+
+__all__ = [
+    "ADE20kDatasetHparams", "ADE20kWebDatasetHparams", "BratsDatasetHparams", "C4DatasetHparams",
+    "CIFAR10DatasetHparams", "CIFAR10WebDatasetHparams", "CIFAR20WebDatasetHparams", "CIFAR100WebDatasetHparams",
+    "COCODatasetHparams", "DataloaderHparams", "WrappedDataLoader", "get_dataset_registry", "EvaluatorHparams",
+    "GLUEHparams", "DatasetHparams", "SyntheticHparamsMixin", "WebDatasetHparams", "Imagenet1kWebDatasetHparams",
+    "ImagenetDatasetHparams", "TinyImagenet200WebDatasetHparams", "LMDatasetHparams", "MNISTDatasetHparams",
+    "MNISTWebDatasetHparams", "MemoryFormat", "SyntheticBatchPairDataset", "SyntheticDataLabelType",
+    "SyntheticDataType", "SyntheticPILDataset"
```

Comment:
For these, and also the above, can we do:
```
__all__ = [
1,
2,
3,
4,
]
```

format?

Reply:
I tried it the suggested way but make style re-formatted it back.